src/coreclr/vm/ceeload.cpp

```diff
@@ -1087,7 +1090,7 @@ void Module::SetDebuggerInfoBits(DebuggerAssemblyControlFlags newBits)
     }
     else
     {
-        if (!g_pConfig->ForceEnc())
+        if (!(g_pConfig->ForceEnc() || (g_pConfig->DebugAssembliesModifiable() && CORDisableJITOptimizations(GetDebuggerInfoBits()))))
```

Comment:
We are adding more condition to disable EnC. What is actually going to enable EnC when this env variable is set?

Reply:
I think this logic can be simplified:
- Delete `    // If the program has the "ForceEnc" env variable set we ensure every eligible
    // module has EnC turned on.
    if (g_pConfig->ForceEnc() && IsEditAndContinueCapable())
        EnableEditAndContinue();
` in `Module::Initialize`
- Delete `DisableEditAndContinue` method
- Enable EnC here for things that need EnC to be enabled.

Reply:
SetDebuggerInfoBits is called from a lot of places including the ICorDebugModule API.  So doesn't the debugger need to set or clear the ENC flag (EnableEditAndContinue and DisableEditAndContinue) depending on the DACF_ENC_ENABLED flag? What I think you are describing will only allow the debugger to enable ENC not disable it.

Something like this in SetDebuggerInfoBits?

```
 BOOL setEnC = IsEditAndContinueCapable() &&
        ((newBits & DACF_ENC_ENABLED) != 0 ||
          g_pConfig->ForceEnc() ||
         (g_pConfig->DebugAssembliesModifiable() && CORDisableJITOptimizations(GetDebuggerInfoBits())));

    if (setEnC)
    {
        EnableEditAndContinue();
    }

```

Reply:
Or should there be the else clause?
```
    if (setEnC)
    {
        EnableEditAndContinue();
    }
    else
    {
        DisableEditAndContinue();
    }
```

Reply:
In the end, I guess you are assuming that debuggers never enable then disable ENC on a module.

Reply:
Yes, that would be odd. I think enabling EnC should be one way operation - once it is enable, it is cannot be disabled.

Reply:
This would change the semantics of a public debugger API if it doesn't allow disabling.  Are we ok with that even though it would be old?

/cc: @noahfalk